### ISSUE TITLE
Notifications View/Filters

### ISSUE THREAD

**Issue by 0x4007** [ubiquity/work.ubq.fi#117]
## Remarks

I'm still debating if we should consolidate all "list like" UIs on here, but I'm optimistic it could make sense to (like leaderboards as well.) If we include all these UIs here, it may make sense to rebrand to os.ubq.fi. Also the title bar from DevPool Directory to UbiquityOS: 
- Directory
- Proposals
- Notifications
- XP Leaderboard 

## Overview

As activity continues to pick up and my notifications become slower to get through, it would be useful to make a simple ui that allows for more intelligent notification filtering. 

### Prioritization Sorting Algorithm

- We need to detect the priority level of the task the notification is associated with.
   - If the notification is not associated with a task, consider not showing it on this view at all. 
   - If the notification is a pull request, it must be linked to a task with a priority label.
- Reverse chronological order, meaning that neglected/old notifications should appear higher than new ones. 
- Top of list is highest priority, then work your way down to low priority. 

### Blocking Tasks

If it thinks without my input the task is blocked, it's a higher priority notification.
- Linkbacks: if the notification is associated with a task with many linkbacks then it should be considered higher priority, it usually means its blocking other pulls/tasks

Perhaps we could have an AI layer on top one day to summarize and/or reorder notifications from most important to least.

### Remarks
- Always exclude draft pulls.

**Comment by 0x4007** [comment]
I think it's best for me to explore this implementation more before funding it because the vision is not super clear

**Comment by 0x4007** [comment]
Looks like GitHub shipped a fix right away

![image](https://github.com/user-attachments/assets/2f806ffc-fd70-4744-a37f-322bda92ef65)

**Comment by gentlementlegen** [comment]
That is nice to lower the noise! It is hard to follow up everything.

**Comment by rndquu** [comment]
> That is nice to lower the noise! It is hard to follow up everything.

The safest bet is to simply reduce the number of tasks each core team member should digest on a daily basis. It would be way simpler if we had dedicated persons per project, like `core_team_member1 => kernel`, `core_team_member2 => ai features,` `core_team_member3 => blockchain`, etc...

**Comment by 0x4007** [comment]
I dislike traditional delegation because interests across projects ebb and flow for any contributor. This should ideally be automatic/seamless. 

As sort of a dynamic solution for what you propose, we can prioritize the notifications from the repos that the contributor normally clicks on. We could make some type of "repo score" saved within `localStorage` 

That way if contributor is working primarily on kernel, then those could appear higher on the list etc. 

We could even split the UI where the top is sort of the "mandatory" notifications to process, and the rest are "optional" but we can change the naming. But the idea is that if you're responsible for some repository, then it can be made clear and at the top of this UI.

**Comment by gentlementlegen** [comment]
Since we started to have recommendations on who would be the most qualified to solve certain tasks, couldn't the same be applied for this? Although we are not such a big team so the same names might come up often.

**Comment by 0x4007** [comment]
Yes we could tap into the same system once its stable in theory @sshivaditya2019 rfc

**Comment by 0x4007** [comment]
Need to do it in a separate repo because it adds way too much complexity. More info https://github.com/ubiquity/work.ubq.fi/pull/126#issuecomment-2417844076
